pkgs/development/haskell-modules/generic-builder.nix

```diff
@@ -55,7 +55,8 @@ in
 , changelog ? null
 , mainProgram ? null
 , doCoverage ? false
-, doHaddock ? !(ghc.isHaLVM or false)
+# See the [HADDOCK_DOCS] note in pkgs/development/compilers/ghc/9.0.2.nix
+, doHaddock ? !(ghc.isHaLVM or false) && !isCross
```

Comment:
I think a better should would be to add an `hasHaddock` or similar passthru arg to `ghc` and use this here.